extensions/sql-database-projects/src/tools/buildHelper.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { promises as fs, existsSync } from 'fs';
+import * as utils from '../common/utils';
+
+const buildDirectory = 'BuildDirectory';
+const buildFiles: string[] = [
+	'Microsoft.Data.SqlClient.dll',
+	'Microsoft.Data.Tools.Schema.Sql.dll',
+	'Microsoft.Data.Tools.Schema.Tasks.Sql.dll',
+	'Microsoft.Data.Tools.Utilities.dll',
+	'Microsoft.SqlServer.Dac.dll',
+	'Microsoft.SqlServer.Dac.Extensions.dll',
+	'Microsoft.SqlServer.TransactSql.ScriptDom.dll',
+	'Microsoft.SqlServer.Types.dll',
+	'System.ComponentModel.Composition.dll',
+	'Microsoft.Data.Tools.Schema.SqlTasks.targets'
+];
+
+export class BuildHelper {
+
+	private extensionDir: string;
+	private extensionBuildDir: string;
+	private initialized: boolean = false;
+
+	constructor() {
+		this.extensionDir = vscode.extensions.getExtension('Microsoft.sql-database-projects')?.extensionPath ?? '';
+		this.extensionBuildDir = path.join(this.extensionDir, buildDirectory);
+	}
+
+	// create build dlls directory
+	// this should not be required. temporary solution for issue #10273
+	public async createBuildDirFolder(): Promise<void> {
+
+		if (this.initialized) {
+			return;
+		}
+
+		if (!existsSync(this.extensionBuildDir)) {
+			await fs.mkdir(this.extensionBuildDir);
+		}
+
+		const buildfilesPath = await this.getBuildDirPathFromMssqlTools();
+
+		buildFiles.forEach(async (fileName) => {
+			if (existsSync(path.join(buildfilesPath, fileName))) {
+				await fs.copyFile(path.join(buildfilesPath, fileName), path.join(this.extensionBuildDir, fileName));
+			}
+		});
```

Comment:
Should this be done asynchronously with `Promise.all()` and the async `utils.exists()`?  `exists()` is coming in from Sakshi's branch, but you can grab the method now: https://github.com/microsoft/azuredatastudio/blob/d246a7deb657f7023d47f51d304baaf9ad1c8e4a/extensions/sql-database-projects/src/common/utils.ts#L65